NorthstarDLL/util/printcommands.cpp

```diff
@@ -160,14 +186,98 @@ void ConCommand_findflags(const CCommand& arg)
 	delete[] upperFlag;
 }
 
+void ConCommand_list(const CCommand& arg)
+{
+	ConCommandBase* var;
+	CCVarIteratorInternal* itint = R2::g_pCVar->FactoryInternalIterator();
+	std::map<std::string, ConCommandBase*> sorted;
+	for (itint->SetFirst(); itint->IsValid(); itint->Next())
+	{
+		var = itint->Get();
+		if (!var->IsFlagSet(FCVAR_DEVELOPMENTONLY) && !var->IsFlagSet(FCVAR_HIDDEN))
+		{
+			sorted.insert({var->m_pszName, var});
+		}
+	}
+	delete itint;
+
+	for (auto& map : sorted)
+	{
+		PrintCommandHelpDialogue(map.second, map.second->m_pszName);
+	}
+	spdlog::info("{} total convars/concommands", sorted.size());
+}
+
+void ConCommand_differences(const CCommand& arg)
+{
+	CCVarIteratorInternal* itint = R2::g_pCVar->FactoryInternalIterator();
+	std::map<std::string, ConCommandBase*> sorted;
+
+	for (itint->SetFirst(); itint->IsValid(); itint->Next())
+	{
+		ConCommandBase* var = itint->Get();
+		if (!var->IsFlagSet(FCVAR_DEVELOPMENTONLY) && !var->IsFlagSet(FCVAR_HIDDEN))
+		{
+			sorted.insert({var->m_pszName, var});
+		}
+	}
+	delete itint;
+
+	for (auto& map : sorted)
+	{
+		ConVar* cvar = R2::g_pCVar->FindVar(map.second->m_pszName);
+
+		if (!cvar)
+		{
+			continue;
+		}
+
+		if (strcmp(cvar->GetString(), "FCVAR_NEVER_AS_STRING") == NULL)
+		{
+			continue;
+		}
+
+		if (strcmp(cvar->GetString(), cvar->m_pszDefaultValue) != NULL)
+		{
```

Comment:
one more guard clause?

Reply:
didn't like the look of the code with that one as a guard clause but sure 
1 sec

Reply:
:+1: 135be3ef1ebfe9bebb8c8978c14cb9d7a70755ca